From 493a45926a50ac9c7033a1e7242edf9b576d8a07 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:55 -0400
Subject: [fs] btrfs: compression - reuse recently used workspace

Message-id: <1407179109-13714-409-git-send-email-sandeen@redhat.com>
Patchwork-id: 87180
O-Subject: [RHEL7.1 PATCH 408/422] btrfs compression: reuse recently used workspace
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Sergey Senozhatsky <sergey.senozhatsky@gmail.com>

commit c39aa7056f50fa7b49265ea35c0f3eddb7c7be68
Author: Sergey Senozhatsky <sergey.senozhatsky@gmail.com>
Date:   Wed Jun 25 00:00:44 2014 +0900

    btrfs compression: reuse recently used workspace

    Add compression `workspace' in free_workspace() to
    `idle_workspace' list head, instead of tail. So we have
    better chances to reuse most recently used `workspace'.

    Signed-off-by: Sergey Senozhatsky <sergey.senozhatsky@gmail.com>
    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/compression.c b/fs/btrfs/compression.c
index 149a331..9aa231d 100644
--- a/fs/btrfs/compression.c
+++ b/fs/btrfs/compression.c
@@ -822,7 +822,7 @@ static void free_workspace(int type, struct list_head *workspace)
 
  spin_lock(workspace_lock);
  if (*num_workspace < num_online_cpus()) {
-  list_add_tail(workspace, idle_workspace);
+  list_add(workspace, idle_workspace);
   (*num_workspace)++;
   spin_unlock(workspace_lock);
   goto wake;
-- 
1.7.1